File: myproject26/myhomework_2.js


// const numbers = [1, 2, 3, 4, 5]


// // 내장함수 map

// // 새로운 assay(배열), 제곱수로 구성된 숫자배열
// function mapper(number) {
//     return number ** 2;
//  }

//  // 간편화 
// const mapper = (number) => {
//     return number ** 2;
// }

// const mapper = (number) => number ** 2;

// const mapper = number => number ** 2;
// //

// const nuew_numbers = numbers.map(mapper);
// console.log(new.nuew_numbers);


// const nuew_numbers = numbers.map(
//     number => number ** 2
//     );
// console.log(new_numbers);


//

//sorted 예시

const numbers = [1, 2, 3, 4, 5]; // [31, 89, 24, 81, 46];

// Array의 sort
function make_random_value(number1, number2) {
    return Math.random();
}

const number1 = numbers.map(
    (number) => ({ number, 기준값: Math.random() }),
);

const number2 = number1.sort(
    (value1, value2) => {
        return value1.기준값 - value2.기준값;
    }
);

const number3 = number2.map(
    (value) => value.number
);

console.log(new_numbers);

// map 예시
const numbers = [1, 2, 3, 4, 5];

// 새로운 Array(배열), 제곱수로 구성된 숫자 배열
// function mapper(number) {
//     return number ** 2;
// }

// const mapper = (number) => {
//     return number ** 2;
// }

// const mapper = (number) => number ** 2;

// const mapper = number => number ** 2;

// const new_numbers = numbers.map(mapper);
// console.log(new_numbers);

const new_numbers = numbers.map(
    number => number ** 2
);
console.log(new_numbers);

File: myproject26/myhomework_2.py
import random

# numbers = [1, 2, 3, 4, 5]

# # 내장함수 map

# def mapper(number):
#     return numbers ** 2

# print(list(map(mapper, numbers)))

# map(list(map(
#     lambda number: number * 2,
#     numbers,
# )))

### 

numbers = [1, 3, 5, 4, 2]

# 내장함수 sorted
# print(sorted(numbers)) # 오름차순 정렬
# print(sorted(numbers), revese=True) # 내림차순 정렬

def make_value(number):
    return number % 10
print(sorted(numbers, key=make_value))  
# # 일의 자리 오름차순   


# 리스트의 sort 
numbers.sort()
numbers.sort(key=make_value)
print(numbers)

numbers.sort(key=lambda number: number % 10)
print(numbers)

def make_random_value(number):
    return random.randint(1, 100)
numbers.sort(key=make_random_value)
print("random values :", numbers)

## map 예시

numbers = [1, 2, 3, 4, 5]

# 내장함수 map
# def mapper(number):
#     return number ** 2
# print(list(map(mapper, numbers)))

print(list(map(
    lambda number: number ** 2,
    numbers,
)))

# new_numbers = []
# for number in numbers:
#     new_numbers.append(number ** 2)

# # list comprehension
# new_numbers = [
#     number ** 2 for number in numbers
# ]


## sort 예시 
import random


numbers = [31, 89, 24, 81, 46]

          # 31, 81, 24, 46, 89   # ASC
          #  1   1   4  6    9

# 1) 내장함수 sorted
print(sorted(numbers))  # 오름차순
print(sorted(numbers, reverse=True))  # 내림차순

def make_value(number):
    return number % 10
print(sorted(numbers, key=make_value))  # 오름차순


# 2) 리스트의 sort
# numbers.sort()
# numbers.sort(key=make_value)

# numbers.sort(key=lambda number: number % 10)
# print(numbers)

def make_random_value(number):
    return random.randint(1, 100)
numbers.sort(key=make_random_value)
print("random values :", numbers)